http/src/main/java/com/redhat/lightblue/client/http/LightblueHttpClient.java

```diff
@@ -119,21 +127,30 @@ public LightblueResponse metadata(LightblueRequest lightblueRequest) {
     @Override
     public LightblueResponse data(LightblueRequest lightblueRequest) {
         LOGGER.debug("Calling data service with lightblueRequest: " + lightblueRequest.toString());
-        try {
-            return callService(new LightblueHttpDataRequest(lightblueRequest)
-                    .getRestRequest(configuration.getDataServiceURI()));
-        } catch (RuntimeException e) {
-            throw new LightblueHttpClientException("Error sending lightblue request: " + lightblueRequest.getBody(), e);
-        }
+        return callService(new LightblueHttpDataRequest(lightblueRequest)
+                .getRestRequest(configuration.getDataServiceURI()));
     }
 
     @Override
     public <T> T data(LightblueRequest lightblueRequest, Class<T> type) throws IOException {
+        LightblueResponse response = data(lightblueRequest);
+        JsonNode responseJson = response.getJson();
+
+        JsonNode objectType = responseJson.get("objectType");
+        if (objectType != null && "error".equalsIgnoreCase(objectType.textValue())) {
+            throw new RuntimeException("Error in lightblue response: " + responseJson.toString());
+        }
+
+        JsonNode responseNode = responseJson.path("processed");
+        if (responseNode instanceof MissingNode || (responseNode instanceof ArrayNode && responseNode.size() == 0 && !type.isArray())) {
+            return null;
+        }
+
         try {
-            LightblueResponse response = data(lightblueRequest);
-            return response.parseProcessed(type);
-        } catch (RuntimeException | LightblueResponseParseException e) {
-            throw new LightblueHttpClientException("Error sending lightblue request: " + lightblueRequest.getBody(), e);
+            return mapper.readValue(responseNode.traverse(), type);
+        } catch (JsonMappingException e) {
+            LOGGER.error("Error parsing lightblue response: " + responseJson.toString(), e);
+            throw new RuntimeException("Error parsing lightblue response: " + responseJson.toString());
         }
```

Comment:
This change in data(LightblueRequest, Class<T>) reverts changes to use new LightblueResponse#parseProceseed method.  Why?